src/Context/QuizContext.jsx

```diff
@@ -5,7 +5,52 @@ const QuizContext = createContext();
 const QuizProvider = ({ children }) => {
   const [state, dispatch] = useReducer(QuizReducer, {
     CurrentQuiz: {},
-    Result: { score: 0, correct: 0, wrong: 0, attempt: 0 },
+    Result: { score: 0, correct_answer: 0, wrong_answer: 0, attempt: 0 },
+    Category_Report: [
+      {
+        category: "History",
+        score: 0,
```

Comment:
i liked how you create different section for storing marks of different categories